addons/point_of_sale/models/pos_session.py

```diff
@@ -2023,7 +2023,7 @@ def get_pos_ui_res_partner_by_params(self, custom_search_params):
     def get_total_discount(self):
         amount = 0
         for line in self.env['pos.order.line'].search([('order_id', 'in', self.order_ids.ids), ('discount', '>', 0)]):
-            original_price = line.price_subtotal_incl / (1 - line.discount / 100)
+            original_price = line.price_subtotal_incl / ((1 - line.discount / 100) or 1)
```

Comment:
if line.discount is 100, we'll need to compute the original_price using something similar to the following:
```py
original_price = line.tax_ids.compute_all(line.price_unit, line.currency_id, line.qty, product=line.product_id, partner=line.order_id.partner_id)['total_included']
```
So instead of having if..else.. to consider discount=100, I guess we can just always solve the original price using the `compute_all` method. 

And in fact, the `amount` below should be computed like so:

```
amount += original_price - line.price_subtotal_incl
```

Reply:
> if line.discount is 100, we'll need to compute the original_price using something similar to the following:
> 
> ```python
> original_price = line.tax_ids.compute_all(line.price_unit, line.currency_id, line.qty, product=line.product_id, partner=line.order_id.partner_id)['total_included']
> ```
> 
> So instead of having if..else.. to consider discount=100, I guess we can just always solve the original price using the `compute_all` method.
> 
> And in fact, the `amount` below should be computed like so:
> 
> ```
> amount += original_price - line.price_subtotal_incl
> ```

Hello,

Changes have been done!

Thank You!

> if line.discount is 100, we'll need to compute the original_price using something similar to the following:
> 
> ```python
> original_price = line.tax_ids.compute_all(line.price_unit, line.currency_id, line.qty, product=line.product_id, partner=line.order_id.partner_id)['total_included']
> ```
> 
> So instead of having if..else.. to consider discount=100, I guess we can just always solve the original price using the `compute_all` method.
> 
> And in fact, the `amount` below should be computed like so:
> 
> ```
> amount += original_price - line.price_subtotal_incl
> ```

Hello,

Changes have been done!

Thank You!